contracts/src/v0.8/automation/chains/OptimismModule.sol

```diff
@@ -16,13 +16,19 @@ contract OptimismModule is ChainModuleBase {
   uint256 private constant FIXED_GAS_OVERHEAD = 60_000;
   uint256 private constant PER_CALLDATA_BYTE_GAS_OVERHEAD = 270;
 
-  function getCurrentL1Fee() external view override returns (uint256) {
-    return OVM_GASPRICEORACLE.getL1Fee(bytes.concat(msg.data, OP_L1_DATA_FEE_PADDING));
+  // @dev This will be updated to use the new function introduced by OP team
+  function getCurrentL1Fee(uint256 dataSize) external view override returns (uint256) {
+    return _getL1Fee(dataSize);
   }
 
   function getMaxL1Fee(uint256 dataSize) external view override returns (uint256) {
+    return _getL1Fee(dataSize);
+  }
+
+  function _getL1Fee(uint256 dataSize) internal view returns (uint256) {
     // fee is 4 per 0 byte, 16 per non-zero byte. Worst case we can have all non zero-bytes.
     // Instead of setting bytes to non-zero, we initialize 'new bytes' of length 4*dataSize to cover for zero bytes.
+    // this is the same as OP.
     bytes memory txCallData = new bytes(4 * dataSize);
     return OVM_GASPRICEORACLE.getL1Fee(bytes.concat(txCallData, OP_L1_DATA_FEE_PADDING));
```

Comment:
I have major concerns about this.
This implementation has the same problem Functions used to have because getting the exact L1 fee costs a very large amount of gas. This is due to the implementation [here](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/vendor/%40eth-optimism/contracts-bedrock/v0.17.3/src/deps/LibZip.sol#L20) which is very computationally intensive. It is called [here](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/vendor/%40eth-optimism/contracts-bedrock/v0.17.3/src/L2/GasPriceOracle.sol#L60) -> which calls [this](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/vendor/%40eth-optimism/contracts-bedrock/v0.17.3/src/L2/GasPriceOracle.sol#L212)

Reply:
The Functions team found we actually saved users a very large amount of gas by simply charging users the upper limit instead of the exact cost. This is because, simply charging users the upper limit ended up being far cheaper than spending the gas to determine the exact cost. See the implementation [here](https://github.com/smartcontractkit/chainlink/pull/13815/files#diff-5c14c7c9823b218dfb846590fb4a157b5d4368fe04472c5cfcd1f240eb1d7892R51) which uses [this](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/vendor/%40eth-optimism/contracts-bedrock/v0.17.3/src/L2/GasPriceOracle.sol#L75). In fact, the OP team specifically added `getL1FeeUpperBound(uint256 dataSize)` for us for this exact reason.

Reply:
Resolving as I see you are addressing this here: https://github.com/smartcontractkit/chainlink/pull/14151/files